[Bug]: DSV4-Flash QLI ops MTE out of range when sequence length exceeds 300K on 950PR

### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
Collecting environment information...
PyTorch version: 2.10.0+cpu
Is debug build: False

OS: Ubuntu 24.04.3 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: 15.0.7
CMake version: version 4.3.2
Libc version: glibc-2.39

Python version: 3.10.20 (main, Mar 11 2026, 17:46:40) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.6.119-49.22.tl4.x86_64-x86_64-with-glibc2.39

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           52 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  384
On-line CPU(s) list:                     0-383
Vendor ID:                               AuthenticAMD
BIOS Vendor ID:                          Advanced Micro Devices, Inc.
Model name:                              AMD EPYC 9K84 96-Core Processor
BIOS Model name:                         AMD EPYC 9K84 96-Core Processor                 Unknown CPU @ 2.6GHz
BIOS CPU family:                         107
CPU family:                              25
Model:                                   17
Thread(s) per core:                      2
Core(s) per socket:                      96
Socket(s):                               2
Stepping:                                1
Frequency boost:                         enabled
CPU(s) scaling MHz:                      73%
CPU max MHz:                             3703.3760
CPU min MHz:                             1500.0000
BogoMIPS:                                5200.34
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good amd_lbr_v2 nopl xtopology nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba perfmon_v2 ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local avx512_bf16 clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin cppc arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif x2avic v_spec_ctrl vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid overflow_recov succor smca fsrm flush_l1d debug_swap
Virtualization:                          AMD-V
L1d cache:                               6 MiB (192 instances)
L1i cache:                               6 MiB (192 instances)
L2 cache:                                192 MiB (192 instances)
L3 cache:                                768 MiB (24 instances)
NUMA node(s):                            2
NUMA node0 CPU(s):                       0-95,192-287
NUMA node1 CPU(s):                       96-191,288-383
Vulnerability Gather data sampling:      Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Vulnerable
Vulnerability Spec store bypass:         Vulnerable
Vulnerability Spectre v1:                Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:                Vulnerable; IBPB: disabled; STIBP: disabled; PBRSB-eIBRS: Not affected; BHI: Not affected
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Vulnerable
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Vulnerable

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] pyzmq==27.1.0
[pip3] torch==2.10.0+cpu
[pip3] torch_npu==2.10.0.post1.dev20260522
[pip3] torchaudio==2.10.0+cpu
[pip3] torchvision==0.25.0+cpu
[pip3] transformers==4.57.6
[pip3] triton-ascend==3.5.1.dev2026050822
[conda] numpy                                       1.26.4                         pypi_0           pypi
[conda] pyzmq                                       27.1.0                         pypi_0           pypi
[conda] torch                                       2.10.0+cpu                     pypi_0           pypi
[conda] torch-npu                                   2.10.0.post1.dev20260522       pypi_0           pypi
[conda] torchaudio                                  2.10.0+cpu                     pypi_0           pypi
[conda] torchvision                                 0.25.0+cpu                     pypi_0           pypi
[conda] transformers                                4.57.6                         pypi_0           pypi
[conda] triton-ascend                               3.5.1.dev2026050822            pypi_0           pypi
vLLM Version: 0.1.dev17052+g9090368b6 (git sha: 9090368b6)
vLLM Ascend Version: 0.16.0rc2.dev896+g2b8739c19 (git sha: 2b8739c19)

ENV Variables:
ATB_OPSRUNNER_KERNEL_CACHE_LOCAL_COUNT=1
ATB_STREAM_SYNC_EVERY_RUNNER_ENABLE=0
ATB_STREAM_SYNC_EVERY_KERNEL_ENABLE=0
ATB_OPSRUNNER_KERNEL_CACHE_GLOABL_COUNT=5
ATB_HOME_PATH=/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1
ASCEND_TOOLKIT_HOME=/usr/local/Ascend/cann-9.1.0
ATB_COMPARE_TILING_EVERY_KERNEL=0
ASCEND_OPP_PATH=/usr/local/Ascend/cann-9.1.0/opp
LD_LIBRARY_PATH=/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1/lib:/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1/examples:/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1/tests/atbopstest:/usr/local/Ascend/cann-9.1.0/lib64:/usr/local/Ascend/cann-9.1.0/lib64/plugin/opskernel:/usr/local/Ascend/cann-9.1.0/lib64/plugin/nnengine:/usr/local/Ascend/cann-9.1.0/opp/built-in/op_impl/ai_core/tbe/op_tiling/lib/linux/x86_64:/usr/local/Ascend/driver/lib64:/usr/local/Ascend/driver/lib64/common:/usr/local/Ascend/driver/lib64/driver:/usr/local/Ascend/cann-9.1.0/devlib
ASCEND_AICPU_PATH=/usr/local/Ascend/cann-9.1.0
ATB_STREAM_SYNC_EVERY_OPERATION_ENABLE=0
ASCEND_HOME_PATH=/usr/local/Ascend/cann-9.1.0
ATB_MATMUL_SHUFFLE_K_ENABLE=1
ATB_WORKSPACE_MEM_ALLOC_ALG_TYPE=1
ATB_SHARE_MEMORY_NAME_SUFFIX=
TORCH_DEVICE_BACKEND_AUTOLOAD=1
TORCHINDUCTOR_CACHE_DIR=/tmp/torchinductor_root
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1


NPU:
+-------------------------------------------------------------------------------------------------+
| npu-smi 25.7.rc1.1                               Version: 25.7.rc1.1                            |
+--------+------------------+---------------+-----------------------------------------------------+
| NPU ID | Name             | Health        | Power(W)    Temp(C)           Hugepages-Usage(page) |
|        |                  | Bus-Id        | NPU Util(%) Memory-Usage(MB)  HBM-Usage(MB)         |
+========+==================+===============+=====================================================+
| 0      | Ascend950PR      | OK            | 202.6       52                0     / 0             |
|        |                  | 0000:03:00.0  | 0           0    / 0          4806  / 114688        |
+========+==================+===============+=====================================================+
| 1      | Ascend950PR      | OK            | 210.2       56                0     / 0             |
|        |                  | 0000:2E:00.0  | 0           0    / 0          4799  / 114688        |
+========+==================+===============+=====================================================+
| 2      | Ascend950PR      | OK            | 206.4       56                0     / 0             |
|        |                  | 0000:1C:00.0  | 0           0    / 0          4798  / 114688        |
+========+==================+===============+=====================================================+
| 3      | Ascend950PR      | OK            | 212.7       53                0     / 0             |
|        |                  | 0000:16:00.0  | 0           0    / 0          4799  / 114688        |
+========+==================+===============+=====================================================+
| 4      | Ascend950PR      | OK            | 323.3       63                0     / 0             |
|        |                  | 0000:9C:00.0  | 100         0    / 0          106888/ 114688        |
+========+==================+===============+=====================================================+
| 5      | Ascend950PR      | OK            | 304.4       62                0     / 0             |
|        |                  | 0000:BB:00.0  | 99          0    / 0          106375/ 114688        |
+========+==================+===============+=====================================================+
| 6      | Ascend950PR      | OK            | 304.8       62                0     / 0             |
|        |                  | 0000:B6:00.0  | 100         0    / 0          106376/ 114688        |
+========+==================+===============+=====================================================+
| 7      | Ascend950PR      | OK            | 293.6       58                0     / 0             |
|        |                  | 0000:84:00.0  | 16          0    / 0          106378/ 114688        |
+========+==================+===============+=====================================================+
+---------------------------+---------------+-----------------------------------------------------+
| NPU ID                    | Process id    | Process name             | Process memory(MB)       |
+===========================+===============+=====================================================+
| No running processes found in NPU 0                                                             |
+===========================+===============+=====================================================+
| No running processes found in NPU 1                                                             |
+===========================+===============+=====================================================+
| No running processes found in NPU 2                                                             |
+===========================+===============+=====================================================+
| No running processes found in NPU 3                                                             |
+===========================+===============+=====================================================+
| 4                         | 2643474       | VLLMAPIServer            | 291                     |
| 4                         | 2643472       | VLLMAPIServer            | 291                     |
| 4                         | 2643475       | VLLMAPIServer            | 291                     |
| 4                         | 2643119       | vllm                     | 291                     |
| 4                         | 2645952       | VLLMWorker_DP            | 100596                  |
| 4                         | 2643473       | VLLMAPIServer            | 291                     |
| 4                         | 2643468       | VLLMEngineCor            | 291                     |
+===========================+===============+=====================================================+
| 5                         | 2643469       | VLLMEngineCor            | 291                     |
| 5                         | 2645954       | VLLMWorker_DP            | 100738                  |
+===========================+===============+=====================================================+
| 6                         | 2645953       | VLLMWorker_DP            | 100676                  |
| 6                         | 2643470       | VLLMEngineCor            | 291                     |
+===========================+===============+=====================================================+
| 7                         | 2645958       | VLLMWorker_DP            | 100738                  |
| 7                         | 2643471       | VLLMEngineCor            | 291                     |
+===========================+===============+=====================================================+

CANN:
package_name=Ascend-cann-toolkit
version=9.1.0
innerversion=V100R001C11B075
compatible_version=[V100R001C15],[V100R001C19],[V100R001C20],[V100R001C21],[V100R001C23],[V100R001C10]
arch=x86_64
os=linux
path=/usr/local/Ascend/cann-9.1.0
```

</details>


### 🐛 Describe the bug

# Scripts
```python
#!/usr/bin/bash

TIMESTAMP=$(date +"%Y%m%d_%H%M%S")
LOG_FILE="logs/test_${TIMESTAMP}.log"

export OMP_PROC_BIND=false
export OMP_NUM_THREADS=8
export PYTORCH_NPU_ALLOC_CONF=expandable_segments:True
export VLLM_ASCEND_ENABLE_FLASHCOMM1=1

export VLLM_USE_V1=1

export HCCL_BUFFSIZE=1024
export VLLM_VERSION="0.20.0"

export USE_MULTI_BLOCK_POOL=1
export USE_MULTI_GROUPS_KV_CACHE=1

MODEL_WEIGHT=/data2/weights/DeepSeek-V4-Flash

vllm serve $MODEL_WEIGHT \
  --max-model-len 512000 \
  --max-num-batched-tokens 4096 \
  --served-model-name ds \
  --gpu-memory-utilization 0.90 \
  --max-num-seqs 128 \
  --pipeline-parallel-size 1 \
  --data-parallel-size 4 \
  --prefill-context-parallel-size 1 \
  --enable-expert-parallel \
  --port 8000 \
  --block-size 128 \
  --enable-chunked-prefill \
  --no-async-scheduling \
  --tokenizer-mode deepseek_v4 \
  --tool-call-parser deepseek_v4 \
  --enable-auto-tool-choice \
  --reasoning-parser deepseek_v4 \
  --enable-prefix-caching \
  --enforce-eager \
  2>&1 | tee "$LOG_FILE"
```

# VLLM bench scripts
```shell
MODEL_BASE=/data2/weights/DeepSeek-V4-Flash
MODEL_NAME=ds
VLLM_IP=0.0.0.0
VLLM_PORT=9000

unset http_proxy
unset https_proxy
unset all_proxy

vllm bench serve \
  --backend openai-chat \
  --model "$MODEL_BASE" \
  --served-model-name "$MODEL_NAME" \
  --host "$VLLM_IP" \
  --port "$VLLM_PORT" \
  --endpoint /v1/chat/completions \
  --dataset-name random \
  --random-input-len 384000 \
  --random-output-len 1 \
  --num-prompts 2 \
  --request-rate inf \
  --trust-remote-code \
  --max-concurrency 1
```

# Error Plog
## Main Error
```text
[ERROR] RUNTIME(2257012,):2026-05-29-08:33:33.316.266 [device_error_proc_c.cc:654]2258274 ProcessDavidStarsCoreErrorInfo:The extend info: errcode:(95) errorStr: The DDR address of the MTE instruction is out of range. subErrType: 0x4.
[ERROR] RUNTIME(2257012,):2026-05-29-08:33:34.095.092 [davinci_kernel_task.cc:1462]2258274 PrintErrorInfoForDavinciTask:[DFX_INFO]Aicore kernel execute failed, device_id=3, stream_id=61, report_stream_id=61, task_id=1252, flip_num=121, fault kernel_name=QuantLightningIndexer_c6b77d40eb8dde553cddd50dae91ec39_1124279332, fault kernel info ext=none, program id=95, hash=17866595142947152801.
```

## Full Plog
[plog.log](https://github.com/user-attachments/files/28717021/plog.log)

--- a/csrc/attention/quant_lightning_indexer/op_host/quant_lightning_indexer_tiling.cpp
+++ b/csrc/attention/quant_lightning_indexer/op_host/quant_lightning_indexer_tiling.cpp
@@ -871,12 +871,14 @@ ge::graphStatus QuantLightningIndexerTiling::DoTiling(QLITilingInfo *tilingInfo)
     constexpr uint32_t V1_DECODE_DATA_NUM = 2;         // Decode每个核需要存储头和尾部两块数据
     constexpr uint32_t S1_BASE_SIZE = 8;               // S1轴基本块的大小
     constexpr uint32_t TOPK_MAX_SIZE = 2048;           // TopK选取个数
+    constexpr uint32_t ASCEND950_S1_BASE_SIZE = 4;     // Ascend 950 S1轴基本块的大小
+    constexpr uint32_t ASCEND950_S2_BASE_SIZE = 128;   // Ascend 950 S2轴基本块的大小
     uint32_t workspaceSize = ascendcPlatform.GetLibApiWorkSpaceSize();
     // 主流程需Workspace大小
     platform_ascendc::SocVersion socVersion_ = ascendcPlatform.GetSocVersion();
     if (socVersion_ == platform_ascendc::SocVersion::ASCEND950) {
-        constexpr uint32_t s1Base = 4;
-        constexpr uint32_t s2Base = 128;
+        constexpr uint32_t s1Base = ASCEND950_S1_BASE_SIZE;
+        constexpr uint32_t s2Base = ASCEND950_S2_BASE_SIZE;
         workspaceSize += s1Base * ((tilingInfo->s2Size + s2Base - 1) / s2Base) * s2Base * sizeof(uint32_t) * aicNum;
     } else {
         uint32_t mm1ResSize = M_BASE_SIZE * S2_BASE_SIZE;

--- a/csrc/attention/quant_lightning_indexer/op_host/quant_lightning_indexer_tiling.cpp
+++ b/csrc/attention/quant_lightning_indexer/op_host/quant_lightning_indexer_tiling.cpp
@@ -873,13 +873,20 @@ ge::graphStatus QuantLightningIndexerTiling::DoTiling(QLITilingInfo *tilingInfo)
     constexpr uint32_t TOPK_MAX_SIZE = 2048;           // TopK选取个数
     uint32_t workspaceSize = ascendcPlatform.GetLibApiWorkSpaceSize();
     // 主流程需Workspace大小
-    uint32_t mm1ResSize = M_BASE_SIZE * S2_BASE_SIZE;
-    workspaceSize += mm1ResSize * MM1_RES_ELEM_SIZE * DOUBLE_BUFFER * aicNum;
-    // Decode流程(LD)需要Workspace大小
-    // 临时存储Decode中间结果大小: 2(头/尾)*8(s1Base)*2(idx/value)*2048(K)*sizeof(int32)*24=6M
-    workspaceSize += V1_DECODE_DATA_NUM * S1_BASE_SIZE * V1_RES_ELEM_TYPE * TOPK_MAX_SIZE * V1_RES_ELEM_SIZE * aicNum;
-    // 临时存储Decode中间参数信息大小: 2(头/尾)*8(s1Base)*16(paramNum)*sizeof(int64_t)*24=48k
-    workspaceSize += V1_DECODE_DATA_NUM * S1_BASE_SIZE * V1_DECODE_PARAM_NUM * V1_DECODE_PARAM_ELEM_SIZE * aicNum;
+    platform_ascendc::SocVersion socVersion_ = ascendcPlatform.GetSocVersion();
+    if (socVersion_ == platform_ascendc::SocVersion::ASCEND950) {
+        constexpr uint32_t s1Base = 4;
+        constexpr uint32_t s2Base = 128;
+        workspaceSize += s1Base * ((tilingInfo->s2Size + s2Base - 1) / s2Base) * s2Base * sizeof(uint32_t) * aicNum;
+    } else {
+        uint32_t mm1ResSize = M_BASE_SIZE * S2_BASE_SIZE;
+        workspaceSize += mm1ResSize * MM1_RES_ELEM_SIZE * DOUBLE_BUFFER * aicNum;
+        // Decode流程(LD)需要Workspace大小
+        // 临时存储Decode中间结果大小: 2(头/尾)*8(s1Base)*2(idx/value)*2048(K)*sizeof(int32)*24=6M
+        workspaceSize += V1_DECODE_DATA_NUM * S1_BASE_SIZE * V1_RES_ELEM_TYPE * TOPK_MAX_SIZE * V1_RES_ELEM_SIZE * aicNum;
+        // 临时存储Decode中间参数信息大小: 2(头/尾)*8(s1Base)*16(paramNum)*sizeof(int64_t)*24=48k
+        workspaceSize += V1_DECODE_DATA_NUM * S1_BASE_SIZE * V1_DECODE_PARAM_NUM * V1_DECODE_PARAM_ELEM_SIZE * aicNum;
+    }
     size_t *workSpaces = context_->GetWorkspaceSizes(1);
     workSpaces[0] = workspaceSize;
 

--- a/csrc/attention/quant_lightning_indexer/op_kernel/arch35/quant_lightning_indexer_kernel.h
+++ b/csrc/attention/quant_lightning_indexer/op_kernel/arch35/quant_lightning_indexer_kernel.h
@@ -398,9 +398,11 @@ __aicore__ inline void QLIPreload<QLIT>::Init(__gm__ uint8_t *query, __gm__ uint
     uint64_t offset = 0;
     //vec 把整个s2的score存储在GM，大小为s1BaseSize * 16K * 4
     GlobalTensor<SCORE_T> scoreGm; //存放vec核写出的score
-    uint64_t singleCoreScoreSize = constInfo.s1BaseSize * QLICommon::Align((uint64_t)constInfo.kSeqSize, (uint64_t)constInfo.s2BaseSize)  * sizeof(SCORE_T);
-    scoreGm.SetGlobalBuffer((__gm__ SCORE_T *)(workspace + aiCoreIdx * singleCoreScoreSize));
-    offset += GetBlockNum() * singleCoreScoreSize;
+    if ASCEND_IS_AIV {
+        uint64_t singleCoreScoreSize = constInfo.s1BaseSize * QLICommon::Align((uint64_t)constInfo.kSeqSize, (uint64_t)constInfo.s2BaseSize)  * sizeof(SCORE_T);
+        scoreGm.SetGlobalBuffer((__gm__ SCORE_T *)(workspace + aiCoreIdx * singleCoreScoreSize));
+        offset += GetBlockNum() * singleCoreScoreSize;
+    }
 
     if ASCEND_IS_AIV {
         vectorService.InitParams(constInfo, tiling);

